lisa/tools/sar.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from decimal import Decimal
+from typing import List, cast
+
+from lisa.executable import Tool
+from lisa.messages import NetworkPPSPerformanceMessage
+from lisa.operating_system import Posix
+from lisa.util.process import Process
+
+
+class Sar(Tool):
+    # 06:37:41        IFACE   rxpck/s   txpck/s    rxkB/s    txkB/s   rxcmp/s   txcmp/s  rxmcst/s   %ifutil # noqa: E501
+    # 06:37:42           lo      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00 # noqa: E501
+    # 06:37:42         eth0   2856.00   2857.00    186.86    187.28      0.00      0.00      0.00      0.00 # noqa: E501
+    #
+    # 06:37:42        IFACE   rxpck/s   txpck/s    rxkB/s    txkB/s   rxcmp/s   txcmp/s  rxmcst/s   %ifutil # noqa: E501
+    # 06:37:43           lo      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00 # noqa: E501
+    # 06:37:43         eth0   3195.00   3194.00    209.04    209.33      0.00      0.00      0.00      0.00 # noqa: E501
+    sar_results_pattern = re.compile(r"(IFACE[\w\W]*?)(?=IFACE|\Z)", re.MULTILINE)
+
+    @property
+    def command(self) -> str:
+        return "sar"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("sysstat")
+        return self._check_exists()
+
+    def run_with_parameter_async(
+        self, key_word: str = "DEV", interval: int = 1, count: int = 300
+    ) -> Process:
+        # sar [ options ] [ <interval> [ <count> ] ]
+        # -n { <keyword> [,...] | ALL }
+        #     Network statistics
+        #     Keywords are:
+        #     DEV     Network interfaces
+        #     EDEV    Network interfaces (errors)
+        #     NFS     NFS client
+        #     NFSD    NFS server
+        #     SOCK    Sockets (v4)
+        #     IP      IP traffic      (v4)
+        #     EIP     IP traffic      (v4) (errors)
+        #     ICMP    ICMP traffic    (v4)
+        #     EICMP   ICMP traffic    (v4) (errors)
+        #     TCP     TCP traffic     (v4)
+        #     ETCP    TCP traffic     (v4) (errors)
+        #     UDP     UDP traffic     (v4)
+        #     SOCK6   Sockets (v6)
+        #     IP6     IP traffic      (v6)
+        #     EIP6    IP traffic      (v6) (errors)
+        #     ICMP6   ICMP traffic    (v6)
+        #     EICMP6  ICMP traffic    (v6) (errors)
+        #     UDP6    UDP traffic     (v6)
+        #     FC      Fibre channel HBAs
+        #     SOFT    Software-based network processing
+        cmd = f"{self.command} -n {key_word} {interval} {count}"
+        process = self.node.execute_async(cmd)
+        return process
+
+    def run_with(
```

Comment:
What means `run_with`? Do you want to say `run_with_paramter`? Because the `run` is used for tool, but "with_parameter" is not meaningful, how about `get_statistics`?

Reply:
updated